auditbeat/module/auditd/audit_linux.go

```diff
@@ -168,10 +175,13 @@ func closeAuditClient(client *libaudit.AuditClient, log *logp.Logger) {
 // kernel until the reporter's done channel is closed.
 func (ms *MetricSet) Run(reporter mb.PushReporterV2) {
 	defer closeAuditClient(ms.client, ms.log)
+	defer ms.control.Close()
```

Comment:
Do you think this should be using the same `closeAuditClient()` to close `ms.control`? Not sure of the history that necessitated closeAuditClient.

Reply:
No, it doesn't need `closeAuditClient`. The [`Close`](https://github.com/elastic/go-libaudit/blob/main/audit.go#L425-L446) function of `AuditClient` only sends a message before closing if [`SetPID`](https://github.com/elastic/go-libaudit/blob/main/audit.go#L310-L320) has been called. Which it is hasn't for `ms.control`. 

From what I can glean from the [deadlock fix](https://github.com/elastic/beats/pull/26032), the message being sent was sometimes being dropped when the netlink socket was overwhelmed.  This led to a deadlock. To remedy the problem, a go routine was started to drain the socket in parallel to the message being sent.

Seems like calling `closeAuditClient` is harmless though. Would you like me to change it?

Reply:
No, thanks for investigating and explaining.